include/freerdp/codec/h264.h

```diff
@@ -74,30 +74,20 @@ extern "C"
 	                                  const RECTANGLE_16* regionRect, BYTE** ppDstData,
 	                                  UINT32* pDstSize, RDPGFX_H264_METABLOCK* meta);
 
-	/** @brief API for user to fill YUV I420 buffer before encoding
-	 *
-	 *  @param h264 The h264 context to query
-	 *  @param nSrcStride The size of a line in bytes of the source image
-	 *  @param nSrcWidth The width of the source image in pixels
-	 *  @param nSrcHeight The height of the source image
-	 *  @param YUVData A pointer to hold the current YUV buffers
-	 *  @param stride A pointer to hold the byte length of a line in the YUV buffers
-	 *  @return \b >= 0 for success, \b <0 for an error
-	 *  @since version 3.6.0
-	 */
-	FREERDP_API INT32 h264_get_yuv_buffer(H264_CONTEXT* h264, UINT32 nSrcStride, UINT32 nSrcWidth,
-	                                      UINT32 nSrcHeight, BYTE* YUVData[3], UINT32 stride[3]);
-
 	/**
 	 * @brief Compress currently filled image data to H264 stream
 	 *
 	 * @param h264 The H264 context to use for compression
+	 * @param pSrcData A pointer to pointers to input buffers (in YUV420P or NV12 format)
+	 * @param pSrcStride A pointer to array of line byte lengths in the input buffers
 	 * @param ppDstData A pointer that will hold the allocated result buffer
 	 * @param pDstSize A pointer for the destination buffer size in bytes
 	 * @return \b >= 0 for success, \b <0 for an error
 	 * @since version 3.6.0
 	 */
-	FREERDP_API INT32 h264_compress(H264_CONTEXT* h264, BYTE** ppDstData, UINT32* pDstSize);
+	FREERDP_API INT32 h264_compress(H264_CONTEXT* h264, const BYTE** WINPR_RESTRICT pSrcData,
```

Comment:
First version of MS-RDPECAM implementation relies on source video buffers for `h264_compress` being allocated in `libfreerdp/codec/h264.c, avc420_ensure_buffer()`. The allocation there (old code) assumes source video format is always YUV420P and has number of planes (3) and stride sizes hardcoded for YUV420P.

With this change, we also have to support NV12 as source video format (because it is being used by hardware encoders). NV12 format has 2 planes and different stride sizes, so the old code can't handle it.

The only solution is to allocate/deallocate source buffers outside of `h264.c` and supply them as input parameters for `h264_compress`. I don't see any problem with that, because:
1. New interface is cleaner, `h264_compress` is now a direct map to `h264->subsystem->Compress()` API.
2. Nothing else inside xfreerdp took advantage of `h264_get_yuv_buffer()` and `h264_compress()` after it was introduced in June, so there's no impact here.

I'm trying to avoid modifying the APIs at all cost, but in this case it wasn't possible.

Reply:
Can't you have `h264_compressEx` with your new function signature and have `h264_compress` that would call it with the correct arguments, this way we don't change the API...

Reply:
@oleg0421 not necessarily unused. we´ve published versions containing these symbols.
you can mark them deprecated so we can ditch them once we make a major version jump again.

Reply:
Done